specs/interop/shared-lockbox.md

```diff
@@ -0,0 +1,340 @@
+# Shared Lockbox
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Design](#design)
+  - [Contracts inroductions](#contracts-inroductions)
+    - [`SharedLockbox`](#sharedlockbox)
+    - [`LiquidityMigrator`](#liquiditymigrator)
+  - [Contracts upgrades](#contracts-upgrades)
+    - [`SuperchainConfig`](#superchainconfig)
+    - [`OptimismPortal`](#optimismportal)
+- [Reference implementation](#reference-implementation)
+  - [`SharedLockbox`](#sharedlockbox-1)
+  - [`LiquidityMigrator`](#liquiditymigrator-1)
+  - [`SuperchainConfig`](#superchainconfig-1)
+  - [`OptimismPortal`](#optimismportal-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+With interoperable ETH, withdrawals may fail if the referenced `OptimismPortal` lacks sufficient ETH
+— making it mandatory to withdraw from the L2 where the ETH was originally deposited.
+The `SharedLockbox` improves the Superchain's interoperable ETH withdrawal user experience and avoid this issue.
+To do so, it unifies ETH L1 liquidity in a single contract, enabling seamless withdrawals of ETH from any OP chain
+in the Superchain, regardless of where the ETH was initially deposited.
+
+## Design
+
+### Contracts inroductions
+
+#### `SharedLockbox`
+
+The `SharedLockbox` contract is designed to manage the unified ETH liquidity for the Superchain.
+It provides two main functions: `lockETH` for depositing ETH into the lockbox,
+and `unlockETH` for withdrawing ETH from the lockbox.
+These functions are called by the `OptimismPortal` contracts to manage the shared ETH liquidity
+when making deposits or finalizing withdrawals.
+
+**Interface and properties:**
+
+**`lockETH`**
+
+Deposits and locks ETH into the lockbox's liquidity pool.
+
+- The function MUST accept ETH.
+- Only authorized `OptimismPortal` addresses MUST be allowed to interact.
+- The function MUST emit the `ETHLocked` event with the correct `portal` and `amount`.
```

Comment:
Perhaps could be good to define what the "correct" portal is despite it being somewhat obvious for maximum clarity